controllers/metric_controller.go

```diff
@@ -70,9 +74,30 @@ func NewMetricReconciler(mgr manager.Manager, clusterConfig cluster.Config, cfg
 // Node object and aims to move the current state of the cluster closer to the desired state.
 func (r *metricReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
 	r.log = r.log.WithValues(MetricController, req.NamespacedName)
-
 	// validate if cluster monitoring is enabled in the operator namespace
-	enabled, err := r.validate(ctx)
+
+	namespace := &v1.Namespace{}
+	if err := r.client.Get(ctx, req.NamespacedName, namespace); err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Namespace deleted; clean up previous state
+			delete(r.prevState, req.Name)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Get the current monitoring label state
+	currentLabelValue := namespace.Labels[monitoringLabel]
+	prevLabelValue, exists := r.prevState[namespace.Name]
+
+	// Skip processing if there's no change
+	if exists && currentLabelValue == prevLabelValue {
+		r.log.Info("No change", "namespace", namespace.Name)
+		return ctrl.Result{}, nil
+	}
+	r.prevState[namespace.Name] = currentLabelValue
```

Comment:
This is duplicating logic in r.validate, which checks if the namespace has the label or not